src/util/HeaderUtil.ts

```diff
@@ -547,3 +552,39 @@ export function parseLinkHeader(link: string | string[] = []): LinkEntry[] {
   }
   return links;
 }
+
+/**
+ * Checks if the value of an HTTP Authorization header matches a specific scheme (e.g. Basic, Bearer, etc).
+ *
+ * @param authorization - The value of the Authorization header (may be undefined).
+ * @param scheme - Name of the authorization scheme (case insensitive).
+ * @returns True if the Authorization header uses the specified scheme, false otherwise.
+ */
+export function matchesAuthorizationScheme(authorization: string | undefined, scheme: string): boolean {
+  const lowerCaseScheme = scheme.toLowerCase();
+  if (!authSchemeRegexCache.has(lowerCaseScheme)) {
+    authSchemeRegexCache.set(lowerCaseScheme, new RegExp(`^${lowerCaseScheme} `, 'ui'));
+  }
+  // Support authorization being undefined (for the sake of usability).
+  return authorization !== undefined && authSchemeRegexCache.get(lowerCaseScheme)!!.test(authorization);
```

Comment:
```suggestion
  return typeof authorization !== 'undefined' && authSchemeRegexCache.get(lowerCaseScheme)!.test(authorization);
```

This is slightly safer since in theory it is possible that someone modifies `undefined` (although probably not really relevant). Or you can do `Boolean(authorization && authSchemeRegexCache.get(lowerCaseScheme)!.test(authorization))` which is also valid.

Only 1 `!` is needed to say that we guarantee the result is not undefined (also see something similar below).